pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -840,8 +841,9 @@ func newNSTmplSetSpec() toolchainv1alpha1.NSTemplateSetSpec {
 func newNSTmplSetWithStatus(username, reason, meessage string) *toolchainv1alpha1.NSTemplateSet {
 	return &toolchainv1alpha1.NSTemplateSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      username,
-			Namespace: "toolchain-member",
+			Name:       username,
+			Namespace:  "toolchain-member",
+			Finalizers: []string{toolchainv1alpha1.FinalizerName},
```

Comment:
If I understand it correctly the finalizers should be added by the primary resource controller. Not by the "parent" controller. In this case it should be done by NSTemplateSet controller. As we do it for UserAccounts: https://github.com/codeready-toolchain/member-operator/

Finalizer is an implementation detail of the recourse reconciler. The clients should not care about that. In other words it's only the NSTemplateSet Controller knows how to terminate the nstemplateset resource.

Reply:
ah right. Let me refactor this part of the PR.

Reply:
done in e3dbfff